### ISSUE TITLE
`chunkFileNames` needs to be case sensitive

### ISSUE THREAD

**Issue by shirotech** [rollup/rollup#3139]
- Rollup Version: 1.21.4
- Operating System (or Browser): Mac
- Node Version: v10.16.3

### How Do We Reproduce?

config
`chunkFileNames: '[name].js',`

### Expected Behavior

It is expected to generate name which is case sensitive.

### Actual Behavior

It is generating `Date.js` and `date.js` the same as `date.js` and it confuses the bundle.

**Comment by lukastaegert** [comment]
Not sure I understand you correctly. At the moment the names ARE case sensitive in that `Date` and `date` are different. You are meaning to say we should make it case-insensitive, i.e. if `Date.js` and `date.js` end up next to each other, we deconflict one of them?

**Comment by shirotech** [comment]
No, I meant that for example Date.js and date.js gets combined to date.js for some reason, so currently, it is *insensitive*. We need it to make it *sensitive* for all cases. I'm not exactly sure what causes this case-insensitivity, but when I remove the option `chunkFileNames` it works fine.

**Comment by lukastaegert** [comment]
As you are on a Mac, I would assume you are on a case-insensitive file system as this is the default. That means that when there is a file on the disk named date.js and you store a file named Date.js, then that will replace the existing file. My guess is that it is not Rollup that is replacing the file but your operating system. The reason is that Rollup is, as far as I know, case-sensitive, which means that it thinks date.js and Date.js are different file names and there is no reason to change one of the names before writing to disk (you could check by inspecting the bundle passed to the generateBundle hook if it has separate entries for the two files). The only people for whom this would be the correct choice, though, would be Linux users and some Mac users that specifically chose a case-sensitive file system, so case-insensitive deduplication by default would definitely be the prudent choice.

**Comment by yesmeck** [comment]
@lukastaegert Consider we have two virtual modules named `\0date.js` and  `\0Date.js`, rollup treats these two modules are different, then writing them to `_virtual/_date.js` and `_virtual/_Date.js`, and on a case-insensitive OS，the later `Date.js` will override `date.js`.

I'm trying to fix this issue https://github.com/rollup/rollup/pull/3317, but don't know why CI are failed, could you have a look.
